_posts/2022-XX-XX-configuring-git-repositories.adoc

```diff
@@ -0,0 +1,267 @@
+
+* Enter `__<bitbucket_server_url>__/plugins/servlet/oauth/access-token` as the *Authorize URL*.
+
+* Check the *Create incoming link* checkbox.
+
+Click *Continue* and enter the following details:
+
+* Paste the content of the *bitbucket_consumer_key* file as the *Consumer Key*.
+
+* Enter *{prod-short}* as the *Consumer name*.
+
+* Paste the content of the *public-stripped.pub* file as the *Public Key*.
+
+Click *Continue*.
+
+This completes your user steps. Additional steps are now required from your {prod-short} administrator.
+
+
+==== Admin side
+
+After you've created the reference on the Git SCM, your {prod-short} administrator will create the following Kubernetes secret.
+
+===== GitHub and GitLab
+
+Creating a secret for GitHub and GitLab by a {prod-short} administrator is similar. The only difference between them is the inclusion of the `scm-server-endpoint` annotation. The {prod-short} administrator must enter the URL of the GitLab server because {prod-short} supports the on-prem GitLab.
+
+The administrator can apply the secret by using `{orch-cli}` on a command line:
+
+[source, bash]
+----
+{orch-cli} apply -f - <<EOF
+kind: Secret
+apiVersion: v1
+metadata:
+  name: __<...>__ <1>
+  namespace: __<...>__ <2>
+  labels:
+    app.kubernetes.io/part-of: che.eclipse.org
+    app.kubernetes.io/component: oauth-scm-configuration
+  annotations:
+    che.eclipse.org/oauth-scm-server: __<...>__ <3>
+    che.eclipse.org/scm-server-endpoint: __<gitlab_server_url>__ <4>
+type: Opaque
+data:
+  id: __<...>__ <5>
+  secret: __<...>__ <6>
+EOF
+----
+<1> Depending on your choice of the Git SCM service provider: `github-oauth-config` or `gitlab-oauth-config`.
+<2> The {prod-short} namespace. The default is `{prod-namespace}`.
+<3> Depending on your choice of the Git SCM service provider: `github` or `gitlab`.
+<4> The GitLab server URL (only for GitLab).
+<5> Base64 encoded: the GitHub OAuth Client ID or GitLab Application ID.
+<6> Base64-encoded: the GitHub OAuth Client Secret or GitLab Client Secret.
+
+
+==== Bitbucket
+
+A secret for Bitbucket differs from the GitHub and GitLab secrets mainly in the `data` section.
+
+The administrator can apply the secret by using `{orch-cli}` on a command line:
+
+[source, bash]
+----
+{orch-cli} apply -f - <<EOF
+kind: Secret
+apiVersion: v1
+metadata:
+  name: bitbucket-oauth-config
+  namespace: __<...>__ <1>
+  labels:
+    app.kubernetes.io/component: oauth-scm-configuration
+    app.kubernetes.io/part-of: che.eclipse.org
+  annotations:
+    che.eclipse.org/oauth-scm-server: bitbucket
+    che.eclipse.org/scm-server-endpoint: __<...>__ <2>
+type: Opaque
+data:
+  private.key: __<...>__ <3>
+  consumer.key: __<...>__ <4>
+EOF
+----
+<1> The {prod-short} namespace. The default is `{prod-namespace}`.
+<2> The Bitbucket server URL.
+<3> Base64-encoded content of the `privatepkcs8-stripped.pem` file.
+<4> Base64-encoded content of the `bitbucket_server_consumer_key` file.
```

Comment:
@xbaran4, I can't find any other mention of `bitbucket_server_consumer_key` in this file. Could it be that `bitbucket_consumer_key` is meant here?
```suggestion
<4> Base64-encoded content of the `bitbucket_consumer_key` file.
```

Reply:
Yes, you are right. Good catch!